### PR TITLE
Add sessions page & configure navbar routing

### PR DESCRIPTION
**Pull request by brittanylau**
### Notion ticket link

[Sessions page shell](https://www.notion.so/uwblueprintexecs/9cb454262ed84c6299c0682c23f92ec5?v=1076bd30711a4a01838becee05f8b5a3&p=6d0068f63e9f4c5e9568e21b6e9d8795)
[Navbar shell](https://www.notion.so/uwblueprintexecs/9cb454262ed84c6299c0682c23f92ec5?v=1076bd30711a4a01838becee05f8b5a3&p=e7b0e9967f984489ae74dad29a566866)

### Implementation description

- created a navbar component that links to the registration & sessions pages
- refactored some routing stuff
  - fixed bug where going to /login directly would break because the Login component checks for a referer that doesn't exist
  - created a ProjectREAD component to house all logged-in pages, and provides the field context to them
- added a sessions page
  - removed session-specific code from the registration component
  - registered the private route in the ProjectREAD component

### Steps to test

1. yarn start
2. verify that:
    - if unauthed, going to / or /sessions redirects to /login, and redirects back to the original url after logging in
    - if authed, going to /login redirects to /
    - navbar links work as expected
    - going to a url that doesn't exist (ex. /avocados) redirects to /login if unauthed, / if authed
    - sessions page:

Screen Shot 2021-04-05 at 7 37 20 PM

### What should reviewers focus on?

- testing functionality & code correctness

### Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] I have run the linter
- [x] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: src/shared/App.jsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+function App() {
+  return (
+    <ThemeProvider theme={theme}>
+      <AuthProvider>
+        <Router>
+          <Switch>
+            <Route exact path="/login" component={Login} />
+            <PrivateRoute
+              exact
+              path={["/", "/sessions"]}
```
Comment (brittanylau) [inline src/shared/App.jsx:25]: i don't love listing the paths here bc it's a bit repetitive of what's in the ProjectREAD component - lmk if there's a better way to do this!

**Comment by jolenezheng** [comment]
Rn when I'm logged in and go to a URL that doesn't exist, it redirects me to /login -> /
Is it possible to have it go straight to  / if authed?

**Review by jolenezheng — COMMENTED** [review]
Everything works as expected!! Left a question somewhere in this PR, otherwise she LGTM 🏄🏻‍♂️

**Comment by brittanylau** [comment]
> Rn when I'm logged in and go to a URL that doesn't exist, it redirects me to /login -> /
> Is it possible to have it go straight to / if authed?

hmm atm we do try to redirect to /, but we have to check if they're logged in first! in `PrivateRoute.jsx`, `useContext` only returns & triggers a re-render a split second after the /login redirect is returned, which is why you see it flash on the screen

not 100% if we can get around that, but we can add a loading state later to indicate that we're actually waiting for a response from firebase. let me know what you think!

File: src/shared/App.jsx (hunk 1 above)
Comment (annoyingpr) [reply src/shared/App.jsx:25]: Just a little confused, why do we need to list the paths in both places?

File: src/shared/PrivateRoute.jsx [hunk 2]
```diff
@@ -1,22 +1,21 @@
...
+      {...rest}
+      render={({ location }) =>
         user ? (
-          <Container>
-            <Component {...props} />
-          </Container>
+          <Component />
         ) : (
-          <Redirect to="/login" />
+          <Redirect to={{ pathname: "/login", state: { referer: location } }} />
```
Comment (annoyingpr) [inline src/shared/PrivateRoute.jsx:18]: For my own understanding, is `referer` the page that the user is coming from?

**Review by danielchen115 — APPROVED** [review]
looks like everything works :D lgtm

File: src/components/registration/RegistrationTable.jsx [hunk 3]
```diff
@@ -1,21 +1,24 @@
...
   responsive: "standard",
+  rowsPerPage: 25,
+  rowsPerPageOptions: [25, 50, 100],
   selectableRows: "none",
 };
 
-function getTableRows(families, extraFields) {
-  if (extraFields.length) {
+function getTableRows(families, extraColumns) {
+  if (extraColumns.length) {
```
Comment (annoyingpr) [inline src/components/registration/RegistrationTable.jsx:18]: Why was this renamed?

**Review by annoyingpr — APPROVED** [review]
LGTM 🚀  Just a couple of clarification q's

File: src/components/registration/RegistrationTable.jsx (hunk 3 above)
Comment (brittanylau) [reply src/components/registration/RegistrationTable.jsx]: ooooops I didn't mean to rename that one 😅 good catch

File: src/shared/PrivateRoute.jsx (hunk 2 above)
Comment (brittanylau) [reply src/shared/PrivateRoute.jsx:18]: yes! if you're interested you can find the docs about it [here](https://developer.mozilla.org/en-US/docs/Web/API/Document/referrer) 😄
